src/core/cdr/src/dds_cdrstream_write.part.c

```diff
@@ -503,45 +508,50 @@ static const uint32_t *dds_stream_write_adrBO (uint32_t insn, DDS_OSTREAM_T * __
   return ops;
 }
 
-static const uint32_t *dds_stream_write_delimitedBO (DDS_OSTREAM_T * __restrict os, const struct dds_cdrstream_allocator * __restrict allocator, const char * __restrict data, const uint32_t * __restrict ops)
+static const uint32_t *dds_stream_write_delimitedBO (DDS_OSTREAM_T * __restrict os, const struct dds_cdrstream_allocator * __restrict allocator, const char * __restrict data, const uint32_t * __restrict ops, enum cdr_data_kind cdr_kind)
 {
   uint32_t offs = dds_os_reserve4BO (os, allocator);
-  if (!(ops = dds_stream_write_implBO (os, allocator, data, ops + 1, false)))
+  if (!(ops = dds_stream_write_implBO (os, allocator, data, ops + 1, false, cdr_kind)))
     return NULL;
 
   /* add dheader, which is the serialized size of the data */
-  *((uint32_t *) (os->x.m_buffer + offs - 4)) = to_BO4u (os->x.m_index - offs);
+  *((uint32_t *) (((struct dds_ostream *)os)->m_buffer + offs - 4)) = to_BO4u (((struct dds_ostream *)os)->m_index - offs);
```

Comment:
This file is now also included for native endianness, so that's the reason for adding the casting:
https://github.com/eclipse-cyclonedds/cyclonedds/blob/a0586d7177355a7af07adf124c1b83785838aec3/src/core/cdr/src/dds_cdrstream.c#L1304

We could introduce specialized wrappers for `ostream` to avoid the casting, but I think it's not worth the effort in this case.